### ISSUE TITLE
CSS-like APIs

### ISSUE THREAD

**Issue by spicydonuts** [slamdata/purescript-css#40]
I'd like to leverage `purescript-css` to restrict a CSS-like API that accepts a couple properties individually (like React inline styles, sort of..).

For example, `spectacle` has many CSS-like properties for theming and arranging slides:
```purescript
  ...
  slide
    [ textAlign "center" ]
    [ text "some centered text" ]
  ...
```

I'm making a more PS-friendly wrapper around the library and would really like to import `CSS.TextAlign` and make my attribute builder `textAlign :: forall a. TextAlign -> Attribute a` (I don't even mind having to write a bit of code to unwrap it) but `purescript-css` doesn't seem to support this use case.  Has this already been discussed?  Or I'm not seeing how to do it 😄 

Mentioning @alexmingoia too, since Pux wraps this library and that's probably the likely context in which this would have been discussed already.

**Comment by garyb** [comment]
I don't think it has ever been discussed really. What's the issue with the current arrangement? That everything ends up as just `CSS`?

It also seems like maybe there's an opportunity for the phantom row stuff like we use for indexed HTML in Halogen... I don't know.

**Comment by spicydonuts** [comment]
Yep, I just want to extract the string value that property _would_ get if the whole CSS file was rendered.  It seems like CSS properties with a finite number of values might be better as sum types.  Something like:
```purescript
toKey Center -- "text-align"
```
```purescript
toValue Center -- "center"
```
```purescript
toProperty Center -- "text-align: center;"
```

**Comment by garyb** [comment]
Ahh I see. Well, I'd definitely like to see more sum types too!
